parsl/utils.py

```diff
@@ -380,3 +381,40 @@ def __exit__(
         exc_tb: Optional[TracebackType]
     ) -> None:
         self.cancel()
+
+
+def _strip_non_alphanumeric(raw_string: str) -> str:
+    return re.sub(r'^[^a-z0-9]+|[^a-z0-9]+$', '', raw_string)
+
+
+def sanitize_dns_label_rfc1123(raw_string: str) -> str:
+    """Convert input string to a valid RFC 1123 DNS label."""
+    # Convert to lowercase and replace invalid characters with hyphen
+    sanitized = re.sub(r'[^a-z0-9]', '-', raw_string.lower())
+
+    # Remove consecutive hyphens
+    sanitized = re.sub(r'-+', '-', sanitized)
+
+    # DNS label cannot exceed 63 characters
+    sanitized = sanitized[:63]
+
+    return _strip_non_alphanumeric(sanitized)
```

Comment:
trim to len 63 *after* removing non-alphanumeric to allow more content to be preserved?